lib/src/service/rule_service.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+
+import 'package:angular/angular.dart';
+import 'package:http/http.dart';
+
+import '../data_model/create_user_request.pb.dart';
+import '../data_model/get_rules_response.pb.dart';
+import '../data_model/rule.pb.dart';
+
+/// A class to interact with the Rules Builder server.
+@Injectable()
+class RuleService {
+  /// The base URL of the Rules Builder server.
+  static const serverUrl =
```

Comment:
seems the web server is calling app engine server to get rules, curious why we don't call _firestoreClient.getUserRule() directly within this file.

Reply:
Currently only the web server interacts with Firestore using the default app engine service account. This is because the server stores all the users' rules, and we don't give the client the app engine service account keys. 

If each user stored their own rules, I guess we could have the Excel client directly call Firestore. But we would have to change the design significantly.